src/algorithms/tracking/gnuradio_blocks/glonass_l1_ca_dll_pll_c_aid_tracking_sc.cc

```diff
@@ -604,7 +604,7 @@ int glonass_l1_ca_dll_pll_c_aid_tracking_sc::general_work(int noutput_items __at
             bool enable_dll_pll;
             if (d_enable_extended_integration == true)
                 {
-                    int64_t symbol_diff = round(1000.0 * ((static_cast<double>(d_sample_counter) + d_rem_code_phase_samples) / static_cast<double>(d_fs_in) - d_preamble_timestamp_s));
+                    int64_t symbol_diff = round(1000.0 * ((static_cast<double>(d_sample_counter) + d_rem_code_phase_samples - d_preamble_timestamp_samples) / static_cast<double>(d_fs_in)));
```

Comment:
this comment is unrelated to the meat of your change (which is significant, but I can't speak to the correctness of it). On the one hand, I do appreciate when people do explicit casting to protect against integer division. On the other, this code base goes overboard with redundant casting. C++ type promotion rules guarantee that once one operand of an expression is double, the result will be double. So this whole thing could be simplified with no casting. But before that, let me suggest that the meaning of this line might be more intuitive if the operands were re-arranged slightly. Consider:
```suggestion
                    int64_t symbol_diff = round(1000.0 * (d_sample_counter - d_preamble_timestamp_samples + d_rem_code_phase_samples ) / d_fs_in);
```
Why is this equivalent? Order of operations says the integer sample counts inside the parentheses are evaluated first, then the multiplication with 1000.0 (double) promotes the numerator to double. Then the division by the sample rate results in a double (regardless of the type of `d_fs_in`).

Anyway, whenever I modify a line like this, I try to make it more readable.

Reply:
There are many similar `static_cast`s, that may be removed.
I think, it would be better to not mix functional changes and cleanup.
Maybe, I'll search for static_cast, carefully check types and remove useless statements... later in a dedicated PR.

Reply:
I understand. Personally, I think such a dedicated PR would be too far-reaching; I prefer to refactor as I go, but I don't fault you for not doing it.